CMakeLists.txt

```diff
@@ -283,9 +287,13 @@ if(BUILD_TESTING)
 endif()
 
 if(CELERITAS_USE_CUDA)
-  # Use host compiler by default to ensure ABI consistency
-  set(CMAKE_CUDA_HOST_COMPILER "${CMAKE_CXX_COMPILER}" CACHE STRING
-    "Set to CMAKE_CXX_COMPILER by Celeritas CMakeLists")
+  if(NOT CMAKE_CUDA_ARCHITECTURES)
+    message(WARNING "The CMAKE_CUDA_ARCHITECTURES variable, necessary "
+      "to specify the GPU target device, is not defined, and the default "
+      "guess may be wrong.  Verify that this value matches your CUDA "
+      "device capability."
```

Comment:
For completeness, note that the default is actually the `NVCC` default (as confirmed by log examination and by the documentation at https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/).  As of `CUDA 12.6` the default is still `sm52`) [So technically `CMake` is guessing "*right*" but the `NVCC` default is inappropriate for any modern GPU]